### ISSUE TITLE
Wrong response object on error handler

### ISSUE THREAD

**Issue by marc-mabe** [tuupola/cors-middleware#30]
Hi,

The response object provided as argument to the error callback has not set the current error state. It's still an empty response with `200 OK`.

We are using [problem details error response](https://tools.ietf.org/html/rfc7807) and want to modify the error with the following callback:
```php
function (ServerRequestInterface $request, ResponseInterface $response, $arguments)
use ($problemResponseFactory) {
    $problemDetailsResponse = $problemResponseFactory->createResponse(
        $request,
        $response->getStatusCode(),
        $arguments['message'] ?? $arguments
    );

    return $response
        ->withBody($problemDetailsResponse->getBody())
        ->withHeader('Content-Type', $problemDetailsResponse->getHeader('Content-Type'));
}
```

But this generates a 500 status as body
```
HTTP/1.1 401 Unauthorized
Server: swoole-http-server
Connection: keep-alive
Content-Type: application/problem+json
Date: Fri, 22 Feb 2019 12:18:44 GMT
Content-Length: 157

{
    "title": "Internal Server Error",
    "type": "https://httpstatus.es/500",
    "status": 500,
    "detail": "CORS requested method is not supported."
}
```

... instead of the expected
```
HTTP/1.1 401 Unauthorized
Server: swoole-http-server
Connection: keep-alive
Content-Type: application/problem+json
Date: Fri, 22 Feb 2019 12:18:44 GMT
Content-Length: 157

{
    "title": "Unauthorized",
    "type": "https://httpstatus.es/401",
    "status": 401,
    "detail": "CORS requested method is not supported."
}
```

**REASON**
1. Your middleware defines the error response AFTER calling the error callback
```php
                return $this->processError($request, $response, [
                    "message" => "CORS request origin is not allowed.",
                ])->withStatus(401);
```

2. The problem details generator expects an error code but instead gets a `200` -> this indicates that something unknown went wrong and a `500 Internal Server Error` is generated - which makes sense

**Woraround**
As a workaround we have to hardcode `401` response status but this should be up to your middleware to define.

**Comment by tuupola** [comment]
Ah yes, 401 should be already set when passing the request object to error handler. Something like:

```php
    $response = (new ResponseFactory)->createResponse(401);
    return $this->processError($response, [
        "message" =>  => "CORS request origin is not allowed",
    ]);
```

Will fix.

**Comment by tuupola** [comment]
Could you test if `error-status-code` fixes your problem?

```
$ composer require tuupola/cors-middleware:dev-error-status-code
```

**Comment by marc-mabe** [comment]
@tuupola yes this is fixing the problem 👍

**Comment by tuupola** [comment]
Fixed in [0.9.3](https://packagist.org/packages/tuupola/cors-middleware#0.9.3). Thanks for the heads up!
